neil

```diff
@@ -379,6 +377,12 @@
    :name
    :target-dir
    :overwrite
+   :dry-run
+
+   ; template deps overrides
+   :local/root
```

Comment:
It's not necessary to explicitly list all the options. Babashka CLI automatically parses all command line flags. The list your using here is for converting options in order to options, using `:cmds-opts`.

Reply:
This is just a note.

Reply:
Makes sense. I updated the `:cmds-opts` to be `[:template :name :target-dir]`. These are the ones that made most sense to me as positional arguments.

For the special named arguments which are not directly supported by `deps-new` (`:git/url`, `:git/sha`, `:git/tag`, `:local/root`), we'll probably want to add those to the `neil help` docs in addition to the existing link to the `deps-new` override options.

Reply:
On line 16 (on main) we already have a bunch of options. I'd prefer if we could align the names of options, even if they are used in different contexts. Since we already have `:sha` , maybe we should re-use that one? Or should we deprecate it and rename it to `:git/sha` (with `:sha` being an `:alias` for `:git/sha`)?

Reply:
@borkdude: Here's the solution I came up with:

- `neil new` will always accept namespaced keys like `:git/sha`
- `neil new` now also supports the existing `:sha` and `:latest-sha` options
  - `:sha` is simply renamed to `:git/sha` before parsing
  - `:latest-sha` is a useful new option that provides an alternative to fetching the latest `:git/tag` (which is the default behavior when neither `:git/tag`, `:git/sha`, nor `:latest-sha` is provided)
- Any options for `neil new` that are not already in `babashka.neil/spec` should be documented specifically for `neil new`. If we find some `neil new` options to be useful elsewhere, we can promote them to `babashka.neil/spec` as needed.

What do you think about this?